test/Common/AutoFixture/SutProvider.cs

```diff
@@ -39,46 +62,69 @@ public SutProvider<TSut> SetDependency(Type dependencyType, object dependency, s
         return this;
     }
 
+    /// <summary>
+    /// Gets a dependency of the sut. Can only be called after the SutProvider has been initialized with <see cref="Create"/>.
+    /// As dependencies are initialized with NSubstitute mocks by default, this is often used to retrieve those mocks in order to
+    /// configure them during the arrange stage, or check received calls in the assert stage.
+    /// </summary>
+    /// <param name="parameterName">An optional parameter name to disambiguate the dependency if there are multiple of the same type. You generally don't need this.</param>
+    /// <typeparam name="T">The type of the dependency you want to get - usually an interface.</typeparam>
+    /// <returns>The dependency.</returns>
     public T GetDependency<T>(string parameterName = "") => (T)GetDependency(typeof(T), parameterName);
-    public object GetDependency(Type dependencyType, string parameterName = "")
+
+    /// <summary>
+    /// An overload for <see cref="GetDependency{T}"/> which takes a runtime <see cref="Type"/> object rather than a compile-time type.
+    /// </summary>
+    private object GetDependency(Type dependencyType, string parameterName = "")
     {
         if (DependencyIsSet(dependencyType, parameterName))
         {
             return _dependencies[dependencyType][parameterName];
         }
-        else if (_dependencies.ContainsKey(dependencyType))
+
+        if (_dependencies.ContainsKey(dependencyType))
         {
             var knownDependencies = _dependencies[dependencyType];
             if (knownDependencies.Values.Count == 1)
             {
                 return _dependencies[dependencyType].Values.Single();
             }
-            else
-            {
-                throw new ArgumentException(string.Concat($"Dependency of type {dependencyType.Name} and name ",
-                    $"{parameterName} does not exist. Available dependency names are: ",
-                    string.Join(", ", knownDependencies.Keys)));
-            }
-        }
-        else
-        {
-            throw new ArgumentException($"Dependency of type {dependencyType.Name} and name {parameterName} has not been set.");
```

Comment:
Changes to code here are just removing redundant `else`s. Hide whitespace changes for a better indication.